src/components/SearchHeader.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import {SearchBar} from './SearchBar';
+import {SearchByToggle} from './SearchByToggle';
+
+import './SearchHeader.scss';
+
+export class SearchHeader extends React.PureComponent {
+  constructor(props) {
```

Comment:
As you're not binding anything you can get rid of constructor declaring your state as a prop.

```javascript
export class SearchHeader extends React.PureComponent {
   state = {
      query: '',
      searchBy: 'title',
   }
   ...awesome stuffs
}
```